src/routes/safe/components/Transactions/TxList/Filter/utils.ts

```diff
@@ -81,15 +81,25 @@ const getTransactionFilter = ({
   execution_date__lte,
   value,
 }: Filter): Partial<IncomingFilter | OutgoingFilter> => {
-  const getLocalStartOfDay = (date: string): number => new Date(date).setHours(0, 0, 0, 0)
-  const getLocalEndOfDay = (date: string): number => new Date(date).setHours(23, 59, 59, 999)
+  const getStartOfDay = (date: string): string => {
+    const now = new Date(date)
+    const local = now.setUTCMinutes(now.getTimezoneOffset())
+    return new Date(local).toISOString()
+  }
+
+  const getEndOfDay = (date: string): string => {
+    const now = new Date(date)
+    const hourOffset = now.getTimezoneOffset() / 60
```

Comment:
You should instead set minutes and multiply by 23 rather than dividing. Some places have just a 30 min offset.

Reply:
It accepts a float. I changed it nonetheless.